ctapipe/tools/stage1.py

```diff
@@ -0,0 +1,767 @@
+    output_filename: str
+        output HDF5 file
+    obs_id: int
+        observation ID
+    subarray:
+        SubarrayDescription to get metadata from
+    writer: HDF5TableWriter
+        output
+    """
+    import uuid
+
+    activity = PROV.current_activity.provenance
+
+    reference = meta.Reference(
+        contact=meta.Contact(
+            name="", email="", organization="CTA Consortium"                                               "Consortium"
+        ),
+        product=meta.Product(
+            description="DL1 Data Product",
+            data_category="S",
+            data_level="DL1",
+            data_association="Subarray",
+            data_model_name="ASWG DL1",
+            data_model_version=DL1_DATA_MODEL_VERSION,
+            data_model_url="",
+            format="hdf5",
+        ),
+        process=meta.Process(_type="Simulation", subtype="", _id=str(obs_id), ),
+        activity=meta.Activity.from_provenance(activity),
+        instrument=meta.Instrument(
+            site="Other", # need a way to detect site...
+            class_="Subarray",
+            type_="unknown",
+            version="unknown",
+            id_=subarray.name,
+        ),
+    )
+
+    # convert all values to strings, since hdf5 can't handle Times, etc.:
+    # TODO: add activity_stop_time?
+    headers = {k:str(v) for k,v in reference.to_dict().items()}
+    meta.write_to_hdf5(headers, writer._h5file)
+
+
+def morphology(geom, image_mask) -> MorphologyContainer:
+    """
+    Compute image morphology parameters
+    Parameters
+    ----------
+    geom: ctapipe.instrument.camera.CameraGeometry
+        camera description
+    image_mask: np.ndarray(bool)
+        image of pixels surviving cleaning (True=survives)
+    Returns
+    -------
+    MorphologyContainer:
+        parameters related to the morphology
+    """
+
+    num_islands, island_labels = number_of_islands(geom=geom, mask=image_mask)
+
+    return MorphologyContainer(num_pixels=image_mask.sum(), num_islands=num_islands)
+
+
+class IntensityContainer(Container):
+    """ Store statistics on the intensity distribution of images """
+
+    max = Field(np.nan, "value of pixel with maximum intensity")
+    min = Field(np.nan, "value of pixel with minimum intensity")
+    mean = Field(np.nan, "mean intensity")
+    std = Field(np.nan, "standard deviation of intensity")
+
+
+def intensity_statistics(image) -> IntensityContainer:
+    """ compute intensity statistics of an image  """
+    return IntensityContainer(
+        max=image.max(), min=image[image > 0].min(), mean=image.mean(), std=image.std()
+    )
+
+
+class ExtendedImageParametersContainer(ImageParametersContainer):
+    """
+    Extra parameters to add to the ImageParametersContainer
+
+    TODO: should eventually just move to ImageParametersContainer.
+    """
+
+    intensity = Field(IntensityContainer(), "intensity statistics")
+    mc_intensity = Field(IntensityContainer(), "MC intensity statistics")
```

Comment:
This is an mc value and should probably not go into a container mixed with non-mc data.